include/rocksdb/env.h

```diff
@@ -594,6 +614,22 @@ class RandomAccessFile {
     return Status::OK();
   }
 
+  // Read a bunch of blocks as described by reqs. The blocks can
+  // optionally be read in parallel. This is a synchronous call, i.e it
+  // should return after all reads have completed. The reads will be
+  // non-overlapping. If the function return Status is not ok, status of
+  // individual requests will be ignored and return status will be assumed
+  // for all read requests. The function return status is only meant for any
+  // any errors that occur before even processing specific read requests
+  virtual Status MultiRead(ReadRequest* reqs, size_t num_reqs) {
```

Comment:
Wondering out loud:
If there is a `ReadRequest` should there also be a `ReadResponse`? In the networking protocols, there is typically a response object associated for every request. Here we are clubbing the response (result) into the request itself. 

Another alternative : `ReadContext`. Looking at Linux async io, they seem to have a single struct called a `iocb` (control block) instead of a request and a response. Even RocksDB get has a GetContext. So, may be we can call it `ReadContext` (or something else) ? 

I was wondering about this aloud with @riversand963 just now, but neither of us had a strong opinion on which one option to choose.

Reply:
Networking protocols are inherently asynchronous and unreliable, so request/response makes sense there. I have not seen a similar pattern in storage APIs. You're right that Linux aio has ```iocb```. The kernel has a ```struct request``` and no corresponding response, so I guess there is precedent for that too.

```GetContext``` doesn't have a ```Status```, so I'm a little hesitant to call it ```ReadContext```.